File: Restaurant/src/main/java/com/ISA/Restaurant/Dto/MenuDto.java
package com.ISA.Restaurant.Dto;

//import com.ISA.Restaurant.Dto.MenuItemDto;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class MenuDto implements Serializable {

    private static final long serialVersionUID = 1L;
    private Long menuId;
    private String menuName;
    private String menuDescription;
    private Boolean active;
    private String coverImageUrl;
//    private List<MenuItemDto> items; // Nested DTO for menu items
}

File: frontend/src/services/dto/menu.dto.ts
export interface MenuDto {
  menuId: number;
  menuName: string;
  menuDescription: string;
  active: boolean;
}

export interface RequestMenuSaveDto {
  menuName: string;
  menuDescription: string;
}

export interface RequestUpdatedMenuDto {
  menuName?: string;
  menuDescription?: string;
  active?: boolean;
}
